### ISSUE TITLE
row level lock hang

### ISSUE THREAD

**Issue by maliangzhu** [greenplum-db/gpdb#17541]
### Greenplum version or build
7.0

### OS version and uname -a
centos 7
 

session1: begin; update t1 set b = 3 where a= 1;
sessoin2: begin; update t1 set b = 3 where a= 2; -- gp7  hang here, while gp6 pass

**Comment by kainwen** [comment]
Have you enabled global deadlock detector for both versions?

**Comment by maliangzhu** [comment]
Yes, I forgot enable GP7 global deadlock. After I enable it, there is another problem:
session1: begin; update t1 set b = 3 where a= 1;
sessoin2: begin; update t1 set b = 3 where a= 2; -- pass when gdd enabled
session1:   update t1 set b = 3 where a= 2; -- timeout and terminated
session2:   update t1 set b = 3 where a= 1; --  timeout and pass

after deadlock timeout ,  
select deadlocks from pg_stat_databases ---  deadlocks count doesn't increase
